llama_index/vector_stores/pinecone.py

```diff
@@ -156,6 +157,28 @@ def _to_pinecone_filter(standard_filters: MetadataFilters) -> dict:
 )
 
 
+def _import_pinecone() -> Any:
```

Comment:
`_import_pinecone` and `is_pinecone_v3`, I stole from [LangChain's PR](https://github.com/langchain-ai/langchain/blob/998324939fd8a77ee22bd46412e735f7e1f59203/libs/community/langchain_community/vectorstores/pinecone.py#L369).